symengine/visitor.h

```diff
@@ -86,17 +86,17 @@ class HasSymbolVisitor : public BaseVisitor<HasSymbolVisitor, StopVisitor>
 
     void bvisit(const Basic &x){};
 
-    bool apply(const Basic &b, const RCP<const Symbol> &x)
+    bool apply(const Basic &b, const Symbol& x)
     {
-        x_ = x;
+        x_ = &x;
```

Comment:
This can become a dangling pointer if `x` ever gets deallocated. But since this visitor is only used from the `has_symbol` function, then this should be safe.